controllers/storagecluster/storagecluster_controller_test.go

```diff
@@ -1119,7 +1143,23 @@ func createFakeStorageClusterReconciler(t *testing.T, obj ...runtime.Object) Sto
 			Phase: rookCephv1.ConditionType(statusutil.PhaseReady),
 		},
 	}
-	obj = append(obj, cbp, cfs)
+	verOdf, _ := semver.Make(getSemVer(ocsversion.Version, 1, true))
```

Comment:
This is not ready for review yet.

Reply:
So, since this is an initialization function that creates the fake `StorageCluster` reconcile, which is required by mostly all other unit test functions, I don't think it's right to `assert` anything in it. In case of error, I can manually fill in this field in some way and that'd be a preferred alternative.

Reply:
@raaizik maybe a panic instead of assert makes more sense here, as the test code cannot execute without a fake client.

Reply:
@nb-ohad again, I think in either proposed way it's better to manually fill in this field w/ its nullified value as can be seen in my suggested change. It might not be possible to sort of cancel out this field's value in such a manner in case of an error, but I think it's not logical to have the whole fake client generation procedure panic because of this specific field.

Reply:
@raaizik my understanding is that if this field does not hold the correct value some tests will fail, which means that the entire test run will fail. In that case doesn't it make sense for the test run to fail early so as to not waste time and compute? lets not forget that these tests are running for each PR issued for this repo and in most cases multiple times per pr